grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -114,8 +114,6 @@ class StockMovementApiController {
     def create(StockMovement stockMovement) {
         // Detect whether inbound or outbound stock movement
         def currentLocation = Location.get(session.warehouse.id)
-
-        stockMovement.sourceType = params.sourceType as RequisitionSourceType
```

Comment:
this is because `parseRequest: true` doesn't work anymore in Grails 3 (or even in 2), so as you said -> changing `params` to `request.JSON` should be done here, but since you are binding the data in the params, you "lose' the access to that and this is expected behavior, so LGTM

Reply:
Boom goes the dynamite. Thanks for the reminder @kchelstowski.